docs/rules/init-declarations.md

```diff
@@ -0,0 +1,54 @@
+# Enforce/Disallow Variable Initializations (init-declarations)
+
+In JavaScript, variables can be assigned during declaration, or at any point afterwards using an assignment statement.
+
+## Rule Details
+
+This rule is aimed at enforcing or eliminating variable initializations during declaration. For example, in the following code, `foo` is initialized during declaration, while `bar` is not.
+
+```js
+var foo = 1;
+var bar;
+
+bar = 2;
+```
+
+This rule aims to bring consistency to variable initializations and declarations.
+
+### Options
+
+This rule is configured by passing in the string `"always"` (the default) to enforce initialization at declaration, or `"never"` to disallow initialization during declaration. This rule applies to `var`, `let`, and `const`, however `"never"` is ignored for `const` variables.
+
+You can configure the rule as follows:
+
+```javascript
+{
+    // (default) All variables must be initialized at declaration
+    "init-declarations": [2, "always"],
+
+    // Variables must not be initialized at declaration
+    "init-declarations": [2, "always"]
+}
+```
+
+When configured with `"always"` (the default), the following patterns are considered warnings:
+
+```js
+function foo() {
+    var bar;
+    let baz;
+    const qux;
+}
+```
+
+The following patterns are not considered warnings:
+
+```js
+function foo() {
+    var bar = 1;
+    let baz = 2;
+    const qux = 3;
+}
+```
+
+When configured with `"never"`, the previous conditions are reversed.
```

Comment:
Please give examples for this option also. It helps people understand it easily.